Добавляет статью про оверлеи и отдельные версии сайтов (#5569)
doka-guide/content: a11y/overlays/index.md

---
title: "`Accessible overlays и отдельные версии сайтов`"

> description: "Можно ли решить проблемы доступности, не внося фундаментальные изменения."

Reviewer: Привет,
```suggestion
description: "Можно ли решить проблемы доступности, не внося фундаментальные изменения?"
```
Кажется в дискрипшине спрятался вопрос )

Author: долго думала как написать, решила уже не застревать с названием)

authors:
  - anastasiayarosh
keywords:
  - a11y
  - overlay
  - plugin
  - отдельная версия для слабовидящих
  - доступность
related:
  - a11y/a11y-html
  - a11y/chto-takoe-a11y
  - a11y/wcag
tags:
  - article
---

## Кратко

> Некоторые считатают, что проблемы, связанные с веб-доступностью, можно решить с помощью специальных программ и плагинов, либо сделав отдельную версию для слабовидящих или незрячих пользователей. Является ли это лучшей практикой и можно ли улучшить доступность, не внося фундаментальные правки, изучим в этой статье.

Reviewer: ```suggestion
Некоторые считают, что проблемы, связанные с веб-доступностью, можно решить с помощью специальных программ и плагинов, либо сделав отдельную версию для слабовидящих или незрячих пользователей. Является ли это лучшей практикой и можно ли улучшить доступность, не внося фундаментальные правки, изучим в этой статье.
```
тут опечатка

Author: спасибо, что заметил!

## Оверлеи
### Определение

> Accessible ovelay - это джаваскрипт, написанный для того, чтобы пофиксить проблемы с доступностью на сайте или в приложении. Оверлей трансформирует уже отрисованное приложение или сайт.

Reviewer: > это джаваскрипт, написанный для того..
может лучше, так:
JavaScript-код, написанный для того..

---
title: "`Accessible overlays и отдельные версии сайтов`"
description: "Можно ли решить проблемы доступности, не внося фундаментальные изменения?"
authors:
  - anastasiayarosh
keywords:
  - a11y
  - overlay
  - plugin
  - отдельная версия для слабовидящих
  - доступность
related:
  - a11y/a11y-html
  - a11y/chto-takoe-a11y
  - a11y/wcag
tags:
  - article
---

## Кратко

Некоторые считают, что проблемы, связанные с веб-доступностью, можно решить с помощью специальных программ и плагинов, либо сделав отдельную версию для слабовидящих или незрячих пользователей. Является ли это лучшей практикой и можно ли улучшить доступность, не внося фундаментальные правки, изучим в этой статье.

## Оверлеи
### Определение

Accessible ovelay - это код, написанный на Javascript. Его используют для того, чтобы пофиксить проблемы с доступностью на сайте или в приложении. Оверлей трансформирует уже отрисованное приложение или сайт.

### История появления оверлеев, принципы работы и эволюция

> Первые инструменты для тестирования доступности веб-сайтов были созданы вскоре после того, как была завершена работа над первой версией `WCAG` (Руководства по обеспечению доступности веб-контента) — основного отраслевого стандарта, определяющего требования к доступности в Интернете. Вскоре после выхода финальной версии `WCAG 1.0` были созданы продукты, которые должны были обеспечивать автоматическое соответствие требованиям. Вскоре компании, продвигающие автоматизированное соблюдение требований, подверглись жёсткой критике со стороны специалистов по доступности за ошибочные действия и преувеличение своей полезности для людей с ограниченными возможностями.

Author: Да, правильнее будет про повышение доступности. Спасибо!

Добавление оверлея на сайт может противоречить предпочтениям конечных пользователей в отношении конфиденциальности и создавать риск несоблюдения применимых правил конфиденциальности. Некоторые оверлеи собирают данные и отправляют их поставщику.

Ни один продукт-накладка на рынке не может сделать веб-сайт полностью соответствующим любому существующему стандарту доступности и, следовательно, не может устранить юридические риски в тех странах и регионах, где обеспечение доступности это обязанность бизнеса.
В маркетинге поставщиков оверлеев часто звучит фраза о том, что `"достаточно одной строки кода"`. Они утверждают, что, добавив одну строку (JavaScript-кода), содержащую код оверлея, сайт станет доступным. Далее, как показано ниже, утверждается, что больше ничего не нужно менять на сайте клиента. Это утверждение было бы верным, если бы продукты делали то, что, по словам продавцов, они делают, но это не так. На самом деле было доказано, что такие продукты не могут устранять проблемы с доступностью в следующих типах контента:

> - Изображения и другой нетекстовый контент

Reviewer 2: Можно уточнить, что именно практически невозможно автоматизировать.

Текстовые альтернативы для нетекстового контента: картинки, кнопки и ссылки без текста, но с иконками. Да, есть  компьютерное зрение, но оно не может понять контекст, в котором находится такой контент. Более того, описание может неправильным или избыточным.

В формах сложно автоматически расставлять подписи к полям, обрабатывать ошибки, писать адекватный текст к ним.

Если используешь Angular или React, например, то результат встраиваемого поверх всего оверлея очень непредсказуемый. И ты не можешь это пофиксить, так как вендор гейткипер.

И ещё есть же требование, чтобы всякие сторонние штуки (Silverlight) на сайте, PDF-ки и другие электронные документы, видео и аудио (кэпшены и транскрипции) были доступными. Оверлеи вообще бессильны в этом случае.

- Формы
- Структура заголовка документа
- Доступность клавиатуры и программное управление фокусом

Настройки размера шрифта часто не увеличиваются до одного и того же уровня. Например, в `Allyable`, `AudioEye` и `UserWay` он увеличивается только до 200%, а в `EqualWeb` и `MaxAccess` — только до 150%. Встроенные функции для Windows и macOS позволяют увеличить его намного сильнее. Ни один из режимов высокой контрастности оверлеев не предлагает такого разнообразия, детализации или возможностей настройки, как в `Windows` или `macOS`.

> В последние годы в странах ЕС и США новые законы и судебные иски привели к широкому признанию того, что большинство частных компаний должны соблюдать эти требования. Для тех, кто их не соблюдает и получает жалобы, штрафы могут быть болезненными. Поэтому предоставление решения этой проблемы — очень выгодная бизнес-модель, особенно если вы можете сделать это быстро, дёшево и с минимальными изменениями в существующем пользовательском интерфейсе. Это ложное обещание, которое дают поставщики оверлеев.

Reviewer 2: Тут можно написать, что оверлеи в странах ЕС официально сайты с оверлеями не соответствуют требованиям стандарта EN 301549, Директивы о веб-доступности (Web Accessibility Directive) и скоро вступающему в силу Европейскому закону о доступности (European Accessibility Act). Последний обязует все компании и организации соответствовать WCAG 2.1 на уровне AA.

> In December 2023, the European Commission made an [official statement on accessibility overlays and EAA compliance](https://www.edf-feph.org/publications/joint-statement-on-accessibility-overlays/):

> Claims that a website can be made fully compliant without manual intervention are not realistic, since no automated tool can cover all the WCAG 2.1 level A and AA criteria. It is even less realistic to expect to detect automatically the additional EN 301549 criteria.

Reviewer 2: А ещё было бы клёво написать абзац про то, что пользователи доступных интерфейсов не в восторге от оверлеев.

>[AccessiBe got banned](https://nfb.org/about-us/press-room/national-convention-sponsorship-statement-regarding-accessibe) in 2021 by the National Federation of the Blind (NFB). A leading advocacy organization for the blind and visually impaired has had enough of their practices and the negative impact on the community.

> Часто утверждается, что повышение доступности веб-сайта — это способ увеличить конверсию, хотя большинство дискуссий на эту тему основаны лишь на предположениях. Никто, ни поставщики наложений, ни кто-либо другой, никогда не предоставлял достаточных доказательств того, что повышение доступности само по себе приведёт к увеличению коэффициента конверсии. Поставщики оверлеев часто преувеличивают количество своих клиентов. Этим они создают впечатление, что их продукты более популярны, чем на самом деле. На самом деле оверлеи не так сильно востребованы на рынке.

Reviewer 2: Может быть хорошей подводкой к статистике по судебным искам в Штатах. Там есть корреляция между сайтами с оверлеями и количеством судебных исков против таких компаний.

[UsableNet's 2024 Digital Accessibility Lawsuit Report](https://3280432.fs1.hubspotusercontent-na1.net/hubfs/3280432/Remediated%20-%202024-Midyear-Report-1.pdf): Over 20% of lawsuits hit sites using widgets. Over the same period, plaintiffs filed 503 lawsuits against websites with active widgets so far in 2024, compared to 449 suits in 2023.

Одно из свежих дел.

> Tribeca Skin Care vs. accessiBe. On June 24, 2024, Tribeca Skin Care, a small dermatology practice in New York City, filed a class action lawsuit against accessiBe. Even though Tribeca used accessiBe’s overlay, a blind person still sued the practice, claiming their website didn’t meet ADA requirements. This case clearly shows the serious risks of relying solely on these tools for web accessibility.

> В России появляются новые законы о цифровой доступности, поэтому есть свои аналоги оверлеев, которые тоже обещают простое и эффективное решение проблем доступности.

Reviewer 2: Если знаешь примеры таких компаний, то можешь привести. И можно уточнить, что за новые законы и что они пока касаются только госресурсов.

> Пока не существует технологии, позволяющей привести уже существующий веб-сайт в полное соответствие с существующими законами и стандартами без изменения дизайна, структуры и функций. Обычно требуется время, деньги и нарушение процесса разработки, чтобы вставить недостающие элементы доступности. Эксперты по доступности предоставляют консультации по этим процессам, помогая компаниям внедрять новые процессы разработки таким образом, чтобы их цифровые технологии становились доступными каждый раз с самого начала.

Reviewer 2: Было бы клёво рассказать почему такой технологии не существует. В чём загвоздка.

Некоторые когнитивные нарушения трудно или невозможно спроектировать/разработать. Некоторые изменения в доступности могут принести пользу одной группе пользователей, но снизить удобство использования для других.

Именно поэтому существуют вспомогательные технологии. При использовании на компьютере пользователя в качестве аппаратного обеспечения, программного обеспечения или альтернативного устройства пользователь с ограниченными возможностями может выбрать необходимую вспомогательную технологию для удовлетворения своих конкретных потребностей. Из-за разнообразия в природе, причинах и тяжести инвалидности, а также из-за того, что у человека может быть несколько видов инвалидности, ни один из представленных на рынке продуктов не может обеспечить полное решение, и ни один из них даже близко не приблизился к этому.

> Эксперты по доступности, правозащитники и конечные пользователи с ограниченными возможностями согласны с тем, что продукты-накладки не подходят для "исправления" недоступных веб-сайтов. Тем не менее, производители этих продуктов продолжают делать ложные заявления о том, что они обладают технологией, позволяющей сделать веб-сайт доступным и устранить риск судебного разбирательства из-за дискриминации по признаку инвалидности.

Reviewer 2: Кажется, повторяет предыдущую мысль, где была списка на письмо.

Более того, ИИ-решения проблем доступности ломают пользовательский опыт, например, могут создать ложные фокусы, чекбоксы и тд.

## Отдельные версии сайтов для слабовидящих
### Определение

Отдельная версия сайта - версия, которая находится по отдельному URL-адресу. Может отличаться от основной наполнением, наличием изображений и тд. Ссылка на такой сайт обозначается иконкой в виде очков, надписей (`AA`, `"Версия для слабовидящих"` и тд.)

### Пример

Пример такого сайта можно увидеть на сайте [Пенсионного фонда России](http://www.pfrf.ru/).

### В чём суть

Особая версия для слабовидящих, как правило, представляет собой настраиваемую панель с выбором шрифтов и их размера, цветовыми схемами и отличным от основного ресурса расположением содержимого. Призвана помочь слабовидящим пользователям, не перегружая их ненужной информацией.

> ### Плюсы

Reviewer 2: P. S. Надо придумать потом более уникальное название, чтобы не дублировался.

### Минусы

Минусов у этого подхода несколько - нужно поддерживать две версии сайта вместо одной, это затратнее. Поэтому существует большая вероятность того, что отдельная версия сайта будет неполной или устаревшей.
Отдельные версии часто не учитывают потребности других нуждающихся в доступности пользователей - тех, у кого особенности цветого восприятия, дальтонизм, дислексия, нейроотличия, астигматизм и тд.
О наличии отдельной версии пользователь должен каким-то образом узнать и смочь открыть её с помощью клавиатуры. Если сайт плохо свёрстан, это может быть проблематичным.
Содержание отдельного сайта более бедное, чем основная страница, скорее всего, будет меньшее количество изображений или изображения будут отсутствовать совсем.
Общее мнение относительно отдельных версий сайтов такое - отдельные версии для каких-то категорий пользователей это сегрегация и дискриминация. В идеале нужно дать пользователям выбирать самим то, какой контент они хотят увидеть и какие настройки использовать. Почитать более подробный разбор проблематики этого вопроса можно в [блоге, посвящённом доступности](https://www.boia.org/blog/should-you-have-a-separate-accessible-website).

> Главная проблема заключается в том, что основной сайт всё так же остаётся недоступным, проблема доступности не решена.

Reviewer 2: Ещё можно упомянуть, что последний гост ([ГОСТ Р 52872-2019](https://protect.gost.ru/document1.aspx?control=31&baseC=6&page=0&month=8&year=2024&search=%D0%93%D0%9E%D0%A1%D0%A2%20%D0%A0%2052872-2019&id=233736)) прямо упоминает, что отдельные версии не нужны, и в целом фокусируется на всех пользователях с инвалидностью, а не только на слепых и слабовидящих как раньше.

> Настоящий стандарт содержит требования к доступности информации, представленной в электронно-цифровой форме, для широкого круга пользователей с ограничениями жизнедеятельности. Требования стандарта учитывают широкий спектр расстройств функций организма и связанные с ними ограничения жизнедеятельности, в том числе: нарушение зрения, нарушение слуха, нарушение опорно-двигательного аппарата, нарушение речи, нарушение ментальной сферы, трудности в обучении и неврологические нарушения. Хотя настоящий стандарт и охватывает большой спектр разнообразных вопросов, он не может удовлетворить потребности людей со всеми видами и степенями ограничений и их сочетаний. Требования стандарта также учитывают возрастные изменения, что позволяет сделать доступным и удобным интерфейс информационных ресурсов для пожилых людей, а зачастую повышает продуктивность пользования ресурсом всех пользователей в целом.

---
title: "`Accessible overlays и отдельные версии сайтов`"

> description: "Можно ли решить проблемы доступности, не внося фундаментальные изменения?"

Reviewer 2: ```suggestion
description: "Можно ли решить проблемы доступности, не внося серьёзных изменений в код?"
```

authors:
  - anastasiayarosh
keywords:
  - a11y
  - overlay
  - plugin
  - отдельная версия для слабовидящих
  - доступность
related:
  - a11y/a11y-html
  - a11y/chto-takoe-a11y
  - a11y/wcag
tags:
  - article
---

## Кратко

> Некоторые считают, что проблемы, связанные с веб-доступностью, можно решить с помощью специальных программ и плагинов, либо сделав отдельную версию для слабовидящих или незрячих пользователей. Является ли это лучшей практикой и можно ли улучшить доступность, не внося фундаментальные правки, изучим в этой статье.

Reviewer 2: ```suggestion
Некоторые считают, что проблемы с веб-доступностью можно решить с помощью специальных программ, плагинов и отдельной версии сайта для слабовидящих и незрячих пользователей. Можно ли улучшить доступность автоматически и насколько это хорошая практика, изучим в статье.
```

## Оверлеи

> Оверлеи(accessibility overlays, accessibility
> widgets или accessibility plugins) - это сторонний Javascript-код, который встраивают на свой сайт. Его используют для того, чтобы пофиксить проблемы с доступностью на сайте или в приложении. Оверлей трансформирует уже отрисованное приложение или сайт, внося правки поверх отрисованных интерфейсов, после загрузки всей страницы.

Reviewer 2: ```suggestion
_Оверлеи (accessibility overlays, widgets или plugins)_ — это сторонний JavaScript-код, который встраивают на сайт. Этот код используют, чтобы исправить проблемы с доступностью сайта или приложения. Оверлей трансформирует отрисованный интерфейс сайта или приложения после полной загрузки.
```

> Первые инструменты для повышения доступности веб-сайтов были созданы вскоре после того, как была завершена работа над первой версией `WCAG` (Руководства по обеспечению доступности веб-контента) — основного отраслевого стандарта, определяющего требования к доступности в Интернете. Вскоре после выхода финальной версии `WCAG 1.0` были созданы продукты, которые должны были обеспечивать автоматическое соответствие требованиям. Вскоре компании, продвигающие автоматизированное соблюдение требований, подверглись жёсткой критике со стороны специалистов по доступности за ошибочные действия и преувеличение своей полезности для людей с ограниченными возможностями.

Reviewer 2: ```suggestion
Первые инструменты для повышения доступности сайтов появились после того, как была завершена работа над первой версией [Руководств по обеспечению доступности веб-контента](/a11y/wcag/) (коротко WCAG). WCAG — основной отраслевой стандарт, определяющий требования к доступности в интернете.

После выхода WCAG 1.0 появились продукты, которые обещали автоматическое соответствие рекомендациям руководств.
```

Reviewer 2: Предлагаю удалить из абзаца мысль о критике таких решений, так как она, кажется, дублирует последний абзац про критику и не так важна.

> Эти ранние продукты почти всегда представляли собой небольшие элементы управления пользовательским интерфейсом, которые просто зачитывали содержимое страницы вслух. Со временем аналогичные продукты стали добавлять `"виджеты"`, которые должны были функционировать как вспомогательные технологии на странице, добавляя элементы управления для изменения функций веб-сайтов, на которых они были развернуты. Эти элементы управления позволяют увеличивать размер шрифта, его цвет и семейство, менять межбуквенное пространство в большую или меньшую сторону. Оверлеи способны менять выравнивание текста, увеличивать интерфейс, включать режим симуляции цветовой слепоты, выделять отдельную строку чтения для возможности чтения текста людей с дислексией и добавлять индикатор фокуса, если он выключен в стилях. А также оверлеи дают возможность изменять контрастность цветов на странице и внешний вид определенных типов контента на странице.

Reviewer 2: ```suggestion
Ранние продукты представляли собой дополнительные элементы управления интерфейсом, которые зачитывали содержимое страницы вслух. Сегодня их сменили оверлеи, которые работают как вспомогательные технологии и дополнительные настройки, встроенные прямо на страницу сайта.

Оверлеи позволяют увеличивать семейство и размер шрифта, изменять цвет и выравнивание текста, настраивать контрастность цветов или межбуквенное пространство (трекинг). Примеры дополнительных функций оверлеев — увеличение элементов интерфейса, режим симуляции дальтонизма, выделение отдельных строк текста для людей с дислексией и синдромом дефицита внимания, добавление индикатора фокуса, если он выключен в стилях. Также оверлеи дают возможность изменить внешний вид некоторых типов содержимого на странице.

![Примеры возможностей оверлеев](images/overlays.png)
```

Reviewer 2: Можешь уточнить в последнем предложении, каких типов контента?

> Примеров оверлеев достаточно много: `AccessiBe`, `Accessiway`, `AudioEye`, `UserWay` и тд.

Reviewer 2: ```suggestion
Несколько примеров оверлеев: AccessiBe, AccessiWay, AudioEye и UserWay.
```

> Производители таких продуктов утверждают, что их продукт может устранять проблемы с качеством кода на сайтах, на которых он используется. И утверждают, что они используют искусственный интеллект для приведения веб-сайтов в соответствие с законами и стандартами, касающимися доступности. Так ли это на самом деле? Попробуем разобраться.

Reviewer 2: ```suggestion
Производители оверлеев обещают автоматически устранить проблемы с качеством дизайна и кода интерфейса, а также привести сайты в соответствие с требованиями законов о веб-доступности. Дополнительно компании утверждают, что оверлеи стали лучше работать благодаря машинному обучению. Попробуем разобраться, так ли это на самом деле.
```

### Плюсы и минусы

> Для начала отметим те небольшие плюсы, которые есть у оверлеев. Оверлеи способны решить небольшое количество проблем доступности - контрастность, величина текста, передвижение по содержимому сайта с помощью клавиатуры.

Reviewer 2: ```suggestion
Для начала отметим плюсы оверлеев, которые у них есть. Оверлеи могут решить небольшое количество проблем с контрастностью, величиной текста и навигацией по содержимому сайта с помощью клавиатуры.
```

> Минусов у оверлеев гораздо больше.
> Прежде всего оверлей это JavaScript - даже работающий продукт может не сработать из-за того, что скрипт не заработал, произошла какая-то ошибка и тд.

Reviewer 2: ```suggestion
Минусов у оверлеев гораздо больше.

Прежде всего оверлей — это JavaScript-код, который может не сработать из-за того, что не загрузился, произошла ошибка и т. д.
```

> Эти заявления были плохо восприняты профессионалами в области доступности. Более 600 человек подписали заявление сообщества против продуктов-наложений под названием [`Overlay Factsheet`](https://overlayfactsheet.com/en/). В список подписавших информационный бюллетень входят эксперты по доступности, защитники прав людей с ограниченными возможностями, юристы, авторы и редакторы технических спецификаций, на которых основан Интернет. И самое важное - заявление подписали пользователи с ограниченными возможностями. Список подписавших включает представителей более чем 16 стран по всему миру.

Reviewer 2: ```suggestion
На заявления производителей оверлеев отреагировали профессионалы в области доступности. Более 600 человек из 16 стран подписали заявление сообщества против оверлеев «[Overlay Factsheet](https://overlayfactsheet.com/en/)». В списке подписавших не только эксперты по доступности, но и защитники прав людей с инвалидностью, юристы, авторы и редакторы технических спецификаций и пользователи с особыми потребностями.
```

Подробнее о ложных утверждениях поставщиков оверлеев можно узнать на [сайте](https://overlayfalseclaims.com).
В последнее время на волне популярности искусственного интеллекта и его производных начали говорить о том, что именно ИИ-продукты могут с лёгкостью решить проблемы доступности.Существует мнение, что оверлеи на основе искусственного интеллекта могут автоматически обнаруживать и устранять проблемы с доступностью, обеспечивая соответствие стандартам. Но эти инструменты обычно могут устранить лишь небольшой процент барьеров доступности, оставляя многие проблемы нерешенными. Они также не обнаруживают более сложные проблемы, такие как неправильный семантический HTML, сложные интерактивные элементы и мультимедийный контент. Хотя инструменты ИИ могут помочь в выявлении основных проблем доступности, они не могут заменить понимание и навыки решения проблем, которые привносят эксперты-люди. Ручное тестирование и исправление имеют решающее значение для обеспечения полной доступности веб-сайта для всех пользователей.

Более того, ИИ-решения проблем доступности ломают пользовательский опыт, например, могут создать ложные фокусы, чекбоксы и тд.

> ## Отдельные версии сайтов для слабовидящих

Reviewer 2: ```suggestion
## Отдельные версии сайтов
```

> Отдельная версия сайта - версия, которая находится по отдельному URL-адресу. Может отличаться от основной наполнением, наличием изображений и тд. Ссылка на такой сайт обозначается иконкой в виде очков, надписей (`AA`, `"Версия для слабовидящих"` и тд.)

Reviewer 2: ```suggestion
_Отдельная версия сайта_ — это одна или несколько страниц, которые находятся по отдельному URL-адресу. Отдельная версия может отличаться от основной наполнением, например, наличием картинок. Ссылка на отдельную версию обычно обозначается иконкой в виде очков или текстом «Аа», «Версия для слабовидящих» и т. д.
```

> ### Пример
> 
> Пример такого сайта можно увидеть на сайте [Омского портала культуры](https://kultura55.ru).

Reviewer 2: ```suggestion
Пример отдельной версии сайта для пользователей со слепотой и слабовидящих можно увидеть на сайте [Омского портала культуры](https://kultura55.ru).
```

> 
> ![обычная версия сайта](images/ordinary-page.png)
> 
> ![версия сайта для слабовидящих](images/page-version-for-weak-vision.png)

Reviewer 2: ```suggestion
![Обычная версия портала культуры](images/ordinary-page.png)

![Отдельная версия портала для слабовидящих](images/page-version-for-weak-vision.png)
```

Reviewer 2: Можешь немного текстом описать основные отличия между скриншотами?

### В чём суть

> Особая версия для слабовидящих, как правило, представляет собой настраиваемую панель с выбором шрифтов и их размера, цветовыми схемами и отличным от основного ресурса расположением содержимого. Призвана помочь слабовидящим пользователям, не перегружая их ненужной информацией.

Reviewer 2: Можно удалить заголовок «В чём суть» и перенести абзац с описанием отдельной версии под определение этой штуки.

### Достоинства и недостатки

> Основной плюс - возможность настроить шрифты и цвета под себя. Шрифты можно выбирать с засечками или без, другого размера, с разными интервалами и отступами. Цветовых схем обычно несколько разных. Также возможна дополнительная тема страницы.

Reviewer 2: ```suggestion
Единственный плюс — возможность настроить шрифты и цвета под себя. Шрифты можно выбирать с засечками или без, другого размера, с разными интервалами и отступами. Кроме нескольких стандартных цветовых схем есть дополнительные темы страницы.
```

### Плюсы и минусы оверлеев

Для начала отметим плюсы оверлеев, которые у них есть. Оверлеи могут решить небольшое количество проблем с контрастностью, величиной текста и навигацией по содержимому сайта с помощью клавиатуры.

Минусов у оверлеев гораздо больше.

Пока не существует технологии, позволяющей привести уже существующий сайт в полное соответствие с существующими законами и стандартами без изменения дизайна, структуры и функций. Оверлеи трудно контролировать после добавления, их поведение непредсказуемо. Их применение не решает проблемы доступности. Изменения не касаются структуры сайта, поэтому он остаётся недоступным по дефолту. Оверлеи не гибкие и не меняют поведение под конкретного пользователя. Скорее всего, оверлеи не будут достаточно обновляться, чтобы соответствовать обновлениям сайта. Обычно требуется время, деньги и нарушение процесса разработки, чтобы вставить недостающие элементы доступности. Эксперты по доступности предоставляют консультации по этим процессам, помогая компаниям внедрять новые процессы разработки таким образом, чтобы их цифровые технологии становились доступными каждый раз с самого начала.

> В маркетинге поставщиков оверлеев часто звучит фраза о том, что `"достаточно одной строки кода"`. Они утверждают, что, добавив одну строку (JavaScript-кода), содержащую код оверлея, сайт станет доступным. Далее, как показано ниже, утверждается, что больше ничего не нужно менять на сайте клиента. Это утверждение было бы верным, если бы продукты делали то, что, по словам продавцов, они делают, но это не так. На самом деле было доказано, что такие продукты не могут устранять проблемы с доступностью в следующих типах контента:
> 
> - Изображения и другой нетекстовый контент (кнопки и ссылки без текста, но с иконками), проблемы с распознаванием контекста изображений в общем
> - Формы, а конкретнее, подписи к полям, обработку ошибок
> - Структура заголовка документа
> - Доступность клавиатуры и программное управление фокусом
> - Видео, аудио, электронные документы (стороннее содержимое сайта)
> 

Reviewer 2: ```suggestion
```

Reviewer 2: Перенесла повыше.

> Получившийся от встраивания оверлея в сайт результат непредсказуем, так как у оверлея негибкая структура.

Reviewer 2: ```suggestion
Оверлеи трудно контролировать после добавления на сайт, а их поведение непредсказуемо из-за негибкой структуры. Дополнительно  оверлеи не будут часто обновляться и учитывать изменения на сайте.
```

> Прежде всего оверлей — это JavaScript-код, который может не сработать из-за того, что не загрузился, произошла ошибка и т. д.
> 
> Добавление оверлея на сайт может противоречить предпочтениям конечных пользователей в отношении конфиденциальности и создавать риск несоблюдения применимых правил конфиденциальности. Некоторые оверлеи собирают данные и отправляют их поставщику. Так как это сторонний код, который встраивается в первоначальный продукт, это создаёт риск кражи данных через потенциальный бэкдор, который либо там торчит из-за ошибок, допущенных во время разработки, либо оставлен вендором специально для своих целей.
> 
> Вендоры могут продавать данные пользователей оверлеев, поэтому появляется риск нарушения Общего регламента по защите персональных данных (`GDPR`) и Закона Калифорнии о защите персональных данных (`CCPA`).

Reviewer 2: ```suggestion
Так как оверлей — это JavaScript-код, он вообще может не сработать из-за того, что не загрузился или по пути произошла ошибка.

Сторонний код на сайте создаёт риск создаёт риск несоблюдения правил хранения и конфиденциальности данных, а ещё не учитывает предпочтения конечных пользователей в объёмах информации, которой они хотят делиться.

Так как поставщики оверлеев могут продавать данные пользователей, появляется риск нарушения Общего регламента по защите персональных данных (GDPR) и Закона Калифорнии о защите персональных данных (CCPA).
```

> Наложение оверлеев не является необходимым и плохо вписывается в стек технологий по нескольким причинам. Оверлеи отвлекают пользователей от текущего процесса. Это дополнительный код, который может появиться в момент, когда пользователь уже начал важные для себя действия. В некоторых случаях оверлеи не позволяют получить доступ к содержимому страницы, поэтому контент недоступен. Оверлеи могут дезориентировать пользователей с слабым зрением, потому что использование увеличительного стекла для увеличения экрана позволит увидеть небольшую часть страницы.  Хотя возможны некоторые автоматические исправления, не следует использовать оверлей в качестве долгосрочного решения. Если пользователю нужен более крупный текст на веб-сайте, ему нужен более крупный текст на всех сайтах и в приложениях. Если пользователю нужен высококонтрастный режим на веб-сайте, такие насройки нужны на всех сайтах и в приложениях. Если пользователю нужно, чтобы текст на веб-сайте читался вслух, это нужно на всех сайтах и в приложениях.

Reviewer 2: ```suggestion
Наложение оверлеев не является необходимым и плохо вписывается в стек технологий по нескольким причинам. Хотя возможны некоторые автоматические исправления, не следует использовать оверлей в качестве долгосрочного решения.
```

Reviewer 2: Можешь чуть раскрыть по каким причинам плохо вписываются в стек?

Также я перенесла часть предложений чуть ниже.

Author: надеюсь я тебя поняла насчёт стека технологий, сейчас допишу

> Утверждение о том, что оверлеи хороши для преодоления барьеров доступности, игнорирует тот факт, что пользователям, которым нужны эти функции на конкретном сайте, они понадобятся для запуска компьютера, открытия браузера и перехода на сайт. Хотя использование оверлея может улучшить соответствие нескольким положениям основных стандартов доступности, полного соответствия с помощью оверлея достичь невозможно.
> 
> Некоторые когнитивные нарушения трудно или невозможно спроектировать/разработать. Некоторые изменения в доступности могут принести пользу одной группе пользователей, но снизить удобство использования для других. Из-за разнообразия в природе, причинах и тяжести инвалидности, а также из-за того, что у человека может быть несколько видов инвалидности, ни один из представленных на рынке продуктов не может обеспечить полное решение, и ни один из них даже близко не приблизился к этому.
> 
> Именно поэтому существуют вспомогательные технологии. При использовании на компьютере пользователя в качестве аппаратного обеспечения, программного обеспечения или альтернативного устройства пользователь с ограниченными возможностями может выбрать необходимую вспомогательную технологию для удовлетворения своих конкретных потребностей.
> 
> Ещё оверлей может усложнять взаимодействие с интерфейсом, так как использует меньшую площадь экрана. Малейший баг в оверлее может заблокировать доступ к нужным опциям, и это трудно решаемая проблема. Оверлеи потребляют ресурсы системы, что на слабых устройствах может приводить к заметному снижению производительности, особенно в требовательных играх.
> 
> Многие добавляемые оверлеями настройки — избыточны и бесполезны. К примеру, режим высокого контраста уже есть в операционках, и пользователи повышенной контрастности УЖЕ включили эту настройку у себя. Всё, что им нужно - семантичная разметка интерфейса, где кнопки — кнопки, как и другие элементы.
> 
> Смысл чёрно-белого режима тоже не ясен. Если человек не различает цвета в принципе, то он уже так видит интерфейс. Самое главное для него — насколько контрастен текст и элементы интерфейса.
> 
> Размер шрифта тоже увеличивается запросто в браузере, а режим экранной лупы есть у... экранной лупы.
> 
> Настройки размера шрифта часто не увеличиваются до одного и того же уровня. Например, в `Allyable`, `AudioEye` и `UserWay` он увеличивается только до 200%, а в `EqualWeb` и `MaxAccess` — только до 150%. Встроенные функции для Windows и macOS позволяют увеличить его намного сильнее. Ни один из режимов высокой контрастности оверлеев не предлагает такого разнообразия, детализации или возможностей настройки, как в `Windows` или `macOS`.

Reviewer 2: ```suggestion
Оверлеи могут дискриминировать пользователей. Если пользователю нужен более крупный текст, он нужен на всех сайтах и во всех приложениях. Когда человек использует режим высокой контрастности, он должен поддерживаться во всех интерфейсах. Если пользователю нужно, чтобы текст на сайте зачитывался вслух, содержимое должно быть слышно везде.

Из-за разнообразия в природе, причинах, степени и видах инвалидности, ни один из представленных на рынке оверлеев не может сделать интерфейс полностью доступным или даже близко приблизиться к этому. Также невозможно учесть все пользовательские предпочтения и особенности. Например, некоторые когнитивные особенности трудно или вообще невозможно решить.

Оверлеи не делают опыт пользователей равным и одинаковым. Одни и те же настройки могут принести пользу одной группе пользователей, но снизить удобство интерфейса для других. Именно поэтому и существуют вспомогательные технологии. Пользователи с инвалидностью могут выбрать подходящую именно им вспомогательную технологию. Это может быть программа, функция в операционной системе или альтернативное устройство.

Многие настройки в оверлеях бесполезны и избыточны. К примеру, режим высокого контраста уже есть в операционных системах и, скорее всего, уже включены. Размер шрифта легко увеличивается или уменьшается во всех браузерах и операционных системах, а режим экранной лупы есть у… экранной лупы. Если пользователи слушают содержимое, то у них есть [скринридеры](/a11y/screenreaders/) и режимы чтения в браузерах.

Часть функций не влияет на пользовательский опыт. К примеру, чёрно-белый режим для людей с дальтонизмом. Пользователям, которые не различает все цвета, важнее контрастность текста и элементов интерфейса, а не цветовая тема.

Утверждение о том, что оверлеи хороши для преодоления барьеров доступности, игнорирует тот факт, что пользователям, которым нужны эти функции на сайте, понадобится запустить компьютер, открыть браузер и перейти на сайт.

Функции оверлеев работают по-разному и не всегда лучше, чем уже имеющиеся в системе или браузере. Размер шрифта часто не увеличивается до одного и того же уровня. Например, в Allyable, AudioEye и UserWay текст увеличивается только до 200%, а в EqualWeb и MaxAccess — до 150%. Встроенные в Windows и macOS функции позволяют увеличить шрифт намного больше. Ни один из режимов высокой контрастности в оверлеях не предлагает такого разнообразия, детализации и гибкости в настройках, как на Windows.

В каких-то случаях оверлеи ухудшают пользовательский опыт. Они отвлекают пользователей от взаимодействия с интерфейсом. К примеру, всплывающее окно может появиться в момент, когда пользователь начал выполнять важные действия. В некоторых случаях оверлеи не позволяют получить доступ к содержимому страницы. Оверлеи могут дезориентировать пользователей со слабым зрением, потому что использование увеличительного стекла для увеличения экрана позволит увидеть небольшую часть страницы. На мобильных версиях сайтов оверлей может занять большую часть экрана, перекрыть нужные элементы и так усложнить взаимодействие с интерфейсом на небольшом устройстве.

Малейший баг в оверлее может заблокировать доступ к нужным опциям, и это трудно решаемая проблема.
```

Reviewer 2: Клёво было бы пояснить, что имеешь в виду здесь:

- какие когнитивные особенности не учесть в оверлеях в принципе;
- что имеешь в виду, когда пишешь про запуск компьютера, открытие браузера и переход на сайт.

Author: второе не совсем понимаю, можешь пояснить?

Reviewer 2: Почему это проблема для пользователей оверлеев?

Author: я там дописала, что они конфликтуют с ассистивными средствами и мешают пользователям. надо ещё что-то?

> Ещё использование оверлеев плохо влияет на производительность сайта, так как это дополнительный код, требующий дополнительного времени на загрузку. Пользовательский опыт от этого ухудшается, как и SEO-метрики. Более медленная скорость загрузки страницы и худшие показатели обработки страницы специальными поисковыми программами - краулерами - просаживают выдачу сайта в поисковой выдаче.

Reviewer 2: ```suggestion
Использование оверлеев плохо влияет на производительность сайта, так как это дополнительный код, требующий дополнительного времени на загрузку. От этого ухудшается пользовательский опыт, как и SEO-метрики. Медленная скорость загрузки страницы и обработки специальными поисковыми программами — краулерами — понижают сайт в поисковой выдаче.
```

---
title: "Оверлеи и отдельные версии сайтов"
description: "Можно ли решить проблемы доступности, не внося серьёзных изменений в код?"
authors:
  - anastasiayarosh
contributors:
  - tatianafokina

> keywords:

Reviewer 2: Как я сама недавно узнала, в ключевые слова надо включать то, чего нет в тексте. Таким образом в поиске по сайту эта статья появится в большем количестве случаев.

Если не приходит в голову ничего уникального, спокойно можешь удалить это поле.

Причём нуждающиеся в доступности пользователи по большей части настроены против оверлеев. Так, например, оверлей `AccessBi` был (запрещён)[https://nfb.org/about-us/press-room/national-convention-sponsorship-statement-regarding-accessibe] Национальной Федерацией Незрячих (`NFB`), как одно из средств негативного влияния на комьюнити.

> Около 20% судебных дел касается сайтов, использующих оверлеи. Эта статистика увеличивается (503 судебных дел в 2024 году по сравнению с 449 в 2023). Подробнее с этой статистикой можно ознакомиться (по ссылке)[https://3280432.fs1.hubspotusercontent-na1.net/hubfs/3280432/Remediated%20-%202024-Midyear-Report-1.pdf].

Reviewer 2: ```suggestion
В Штатах около 20% судебных исков касается сайтов с оверлеями, и их количество увеличивается. Например, в в 2023 году открыто 449 судебных дел,  а 2024 году — 503. Больше статистики по искам в Штата найдёте на [UsableNet](https://info.usablenet.com/ada-website-compliance-lawsuit-tracker).
```

На громкие заявления производителей оверлеев плохо отреагировало профессиональное сообщество. Более 600 человек из 16 стран подписали заявление против оверлеев «[Overlay Factsheet](https://overlayfactsheet.com/en/)». В списке подписавших не только эксперты по доступности, но и защитники прав людей с инвалидностью, юристы, авторы и редакторы технических спецификаций и пользователи с особыми потребностями.

> Эксперты по доступности, правозащитники и конечные пользователи с ограниченными возможностями согласны с тем, что продукты-накладки не подходят для "исправления" недоступных веб-сайтов. Тем не менее, производители этих продуктов продолжают делать ложные заявления о том, что они обладают технологией, позволяющей сделать веб-сайт доступным и устранить риск судебного разбирательства из-за дискриминации по признаку инвалидности.

Reviewer 2: Этот абзац можно целиком удалить. Мысль уже кажется раскрытой.

Существует целая индустрия консультантов и специалистов по доступности, которые помогают организациям сделать веб-сайты более доступными с помощью законных средств. Организации тратят деньги на исправление недоступного контента в виде оплаты услуг экспертов, времени на проведение тщательной работы и внутренних ресурсов для корректировки процессов разработки и тестирования, которые ранее не учитывали доступность. Ложные заявления поставщиков оверлеев (например, о том, что решения дешевы, быстры и не требуют больших усилий) подрывают законные методы работы консультантов по доступности. Это нечестная конкуренция.

Подробнее о ложных утверждениях поставщиков оверлеев можно узнать на [сайте](https://overlayfalseclaims.com).

> В последнее время на волне популярности искусственного интеллекта и его производных начали говорить о том, что именно ИИ-продукты могут с лёгкостью решить проблемы доступности.Существует мнение, что оверлеи на основе искусственного интеллекта могут автоматически обнаруживать и устранять проблемы с доступностью, обеспечивая соответствие стандартам. Но эти инструменты обычно могут устранить лишь небольшой процент барьеров доступности, оставляя многие проблемы нерешенными. Они также не обнаруживают более сложные проблемы, такие как неправильный семантический HTML, сложные интерактивные элементы и мультимедийный контент. Хотя инструменты ИИ могут помочь в выявлении основных проблем доступности, они не могут заменить понимание и навыки решения проблем, которые привносят эксперты-люди. Ручное тестирование и исправление имеют решающее значение для обеспечения полной доступности веб-сайта для всех пользователей.

Reviewer 2: ```suggestion
На волне популярности искусственного интеллекта и машинного обучения, в сообществе начали говорить о том, что ИИ-продукты могут с лёгкостью решить проблемы доступности. Оверлеи подхватили идею и теперь обещают автоматически обнаруживать и устранять проблемы с доступностью на основе искусственного интеллекта, чтобы сайты соответствовали законам и стандартам.

Хотя инструменты с элементами искусственного интеллекта могут помочь выявить основные проблемы с доступностью, они не могут заменить экспертов с их пониманием контекста и навыками решения проблем. Так что обычно такие инструменты могут устранить лишь небольшой процент барьеров, оставляя многие проблемы нерешёнными. Продвинутые версии оверлеев также не обнаруживают сложные проблемы c HTML, комплексными интерактивными элементами и мультимедийным контентом. Ручное тестирование и исправление всё ещё имеют решающее значение для обеспечения полной доступности сайта для пользователей.
```

> Более того, ИИ-решения проблем доступности ломают пользовательский опыт, например, могут создать ложные фокусы, чекбоксы и тд.

Reviewer 2: ```suggestion
```

Reviewer 2: Перенесла выше.

Особая версия страницы, как правило, — панель с настройками для выбора шрифта и его размера, изменения цветовых схем. Также на отдельной версии сайта отличное от оригинальной страницы расположение содержимого. Это призвано помочь слабовидящим пользователям избежать перегрузки ненужной информацией.

Пример отдельной версии сайта для пользователей со слепотой и слабовидящих можно увидеть на сайте [Омского портала культуры](https://kultura55.ru).

На первом скриншоте обычная версия сайта с цветными изображениями.

![Обычная версия портала культуры](images/ordinary-page.png)

На второй картинке отдельная версия для слабовидящих. На странице нет изображений, шрифт увеличен и установлена чёрно-белая цветовая тема. Дополнительно в отдельной версии можно включить или отключить отрисовку изображений, поменять цветовую тему сайта, величину шрифта и межбуквенное пространство, выбрать тип шрифта (с засечками или без) и изменить межстрочный интервал.

![Отдельная версия портала](images/page-version-for-weak-vision.png)

### Достоинства и недостатки

> Единственный плюс отдельных версий сайтов — возможность настроить шрифты и цвета под себя. Шрифты можно выбирать с засечками или без, разных размеров, с разными интервалами и отступами. Кроме нескольких стандартных цветовых схем есть дополнительные темы страницы.
> 
> Всё это уже могут делать и делают пользователи через настройки браузеров или с помощью плагинов.

Reviewer 2: ```suggestion
Единственный плюс отдельных версий сайтов — возможность настроить шрифты и цвета под себя. Шрифты можно выбирать с засечками или без, разных размеров, с разными интервалами и отступами. Кроме нескольких стандартных цветовых схем есть дополнительные темы страницы. Плюс может стать минусом: пользователи уже всё это могут изменить в настройках браузеров или с помощью расширений и плагинов.
```

Reviewer 2: Объединила и попыталась логически связать.

> Минусов у этого подхода несколько - нужно поддерживать две версии сайта вместо одной, это затратнее. Поэтому существует большая вероятность того, что отдельная версия сайта будет неполной или устаревшей.

Reviewer 2: ```suggestion
Минусов у отдельных версий сайта несколько: нужно поддерживать две версии сайта вместо одной, а это затратно. Так что существует вероятность того, что отдельная версия будет устаревшей.

Также содержимое отдельной страницы не такое полное, как у основной. К примеру, в отдельной версии сайта будет меньше изображений или не быть совсем.
```

> Отдельные версии часто не учитывают потребности других нуждающихся в доступности пользователей - тех, у кого особенности цветого восприятия, дальтонизм, дислексия, нейроотличия, астигматизм и тд.

Reviewer 2: ```suggestion
```

Reviewer 2: Перенесла пониже.

О наличии отдельной версии пользователь должен каким-то образом узнать и смочь открыть её с помощью клавиатуры. Если сайт плохо свёрстан, это может быть проблематичным.

> Содержание отдельного сайта более бедное, чем основная страница, скорее всего, будет меньшее количество изображений или изображения будут отсутствовать совсем.

Reviewer 2: ```suggestion
```

Reviewer 2: Перенесла повыше.

> Общее мнение относительно отдельных версий сайтов такое - отдельные версии для каких-то категорий пользователей это сегрегация и дискриминация. В идеале нужно дать пользователям выбирать самим то, какой контент они хотят увидеть и какие настройки использовать. Почитать более подробный разбор проблематики этого вопроса можно в [блоге, посвящённом доступности](https://www.boia.org/blog/should-you-have-a-separate-accessible-website). Главная проблема заключается в том, что основной сайт всё так же остаётся недоступным, проблема доступности не решена.
> 
> В России появляются новые законы о цифровой доступности, поэтому есть свои аналоги оверлеев, которые тоже обещают простое и эффективное решение проблем доступности. Прежде всего это специальные плагины и скрипты в конструкторах сайтов, например, для `Wordpress`, модуль для сайтов
> "Мибок: Версия для слабовидящих" и тд.
> 
> Новый закон касается только государственных ресурсов, это дополнение к "Федеральному закону от 24.11.1995 N 181-ФЗ (ред. от 29.05.2024)" -
> "О социальной защите инвалидов в Российской Федерации".
> 
> Последний гост `(ГОСТ Р 52872-2019)` прямо поясняет, что отдельные версии сайтов не нужны.  В целом его текст фокусируется на всех пользователях с инвалидностью, а не только на слепых и слабовидящих, как раньше.

Reviewer 2: ```suggestion
Главная проблема отдельных версий заключается в том, что основной сайт всё так же остаётся недоступным, проблема доступности не решена.

Отдельные версии не учитывают потребности всех пользователей, нуждающихся в доступности. К примеру, людей с особенностями цветого восприятия, астигматизмом, дислексией и другими нейроотличиями.

Отдельные версии для определённых категорий пользователей — это сегрегация и дискриминация. Последний гост 52872-2019 прямо говорит, что отдельные версии сайтов не нужны, так как они доступны _только_ для людей со слепотой и слабовидящих.

В идеальной ситуации пользователи должны сами выбирать то, какое содержимое они хотят увидеть и какие настройки использовать. Почитать подробнее про то, в чём проблема разных версий одной страницы, можно в [блоге, посвящённом доступности](https://www.boia.org/blog/should-you-have-a-separate-accessible-website).

В России появляются новые законы о цифровой доступности. Например, дополненный в мае 2024 года Федеральный закон 181-ФЗ «О социальной защите инвалидов в Российской Федерации». Хотя закон касается только государственных сайтов и электронных сервисов, на рынке всё равно появились аналоги оверлеев, которые обещают просто и эффективно решить проблемы с доступностью. Прежде всего это плагины и скрипты в конструкторах сайтов. Например, модуль «Мибок: Версия для слабовидящих» в WordPress. Этот модуль автоматически генерирует отдельную версию сайта.
```

Reviewer 2: Я бы заменила ссылочку на блог Boia. Эти чуваки связаны с AudioEye. Можно вот на эту заменить: 
https://sheribyrnehaber.medium.com/accessibility-separate-but-equal-is-never-ok-e6e97d893d11

> ### Выводы

Reviewer 2: ```suggestion
## Выводы
```

> Попытки решить проблему доступности с помощью оверлеев или отдельных версий сайтов вполне понятны - доступность проще и экономичнее вносить вначале. При этом готовые решения и попытки автоматизации решения проблем доступности не обеспечивают потребности всех нуждающихся в доступности пользователей. Доступность нужна не только незрячим и слабовидящим пользователям, а многим людям ситуативно. Также доступность нужна при когнитивных особенностях или особенностях зрения. Поэтому сайт или приложение необходимо адаптировать под разные нужды, закладывая фундамент доступного сайта начиная от доступного дизайна и заканчивая тщательным тестированием.

Reviewer 2: ```suggestion
Попытки решить проблему доступности интерфейсов с помощью оверлеев или отдельных версий сайтов понятны, но закладывать доступность проще и выгоднее в начале разработки, начиная с дизайна и заканчивая тщательным тестированием.

Готовые и автоматические решения не удовлетворяют потребности всех пользователей, нуждающихся в доступности. Доступные интерфейсы нужны не только незрячим и слабовидящим пользователям, а также при других особенностях зрения, когнитивных или ментальных особенностях. Многие люди сталкиваются с барьерами в вебе ситуативно. Например, при переломе руки или ярком освещении. Сайты и приложения важно адаптировать под разные потребности их пользователей.
```

Reviewer 2: Подвигала предложения.

Многие добавляемые оверлеями настройки — избыточны и бесполезны. К примеру, режим высокого контраста уже есть в операционках, и пользователи повышенной контрастности УЖЕ включили эту настройку у себя. Всё, что им нужно - семантичная разметка интерфейса, где кнопки — кнопки, как и другие элементы.

Смысл чёрно-белого режима тоже не ясен. Если человек не различает цвета в принципе, то он уже так видит интерфейс. Самое главное для него — насколько контрастен текст и элементы интерфейса.

Размер шрифта тоже увеличивается запросто в браузере, а режим экранной лупы есть у... экранной лупы.

Настройки размера шрифта часто не увеличиваются до одного и того же уровня. Например, в `Allyable`, `AudioEye` и `UserWay` он увеличивается только до 200%, а в `EqualWeb` и `MaxAccess` — только до 150%. Встроенные функции для Windows и macOS позволяют увеличить его намного сильнее. Ни один из режимов высокой контрастности оверлеев не предлагает такого разнообразия, детализации или возможностей настройки, как в `Windows` или `macOS`.

> Использование оверлеев плохо влияет на производительность сайта, так как это дополнительный код, требующий дополнительного времени на загрузку. От этого ухудшается пользовательский опыт, как и SEO-метрики. Медленная скорость загрузки страницы и обработки специальными поисковыми программами — краулерами — понижают сайт в поисковой выдаче.

Reviewer 2: ```suggestion
Использование оверлеев плохо влияет на производительность сайта, так как это дополнительный код, требующий дополнительного времени на загрузку. От этого ухудшается пользовательский опыт, как и SEO-метрики. Медленная скорость загрузки страницы и обработки специальными поисковыми программами — краулерами — понижают сайт в поисковой выдаче.

Оверлеи потребляют ресурсы системы, что на слабых устройствах может приводить к заметному снижению производительности, особенно в требовательных играх.
```

Reviewer 2: Это частичка другого абзаца, которая хорошо подходит к теме производительности. Только можешь пояснить, как оверлеи влияют на игры?

![Главная страница нидерландского государственного сайта с включённым режимом чтения содержимого](images/reading-overlay.png)

Оверлеи позволяют увеличивать размер шрифта, изменять цвет и выравнивание текста, настраивать контрастность цветов или межбуквенное пространство. Примеры дополнительных функций оверлеев — увеличение элементов интерфейса, режим симуляции дальтонизма, выделение отдельных строк текста для людей с дислексией, добавление отсутствующего индикатора фокуса и пр.

![Некоторые функции и возможности оверлеев](images/overlays.png)

Поставщики AccessiBe, AccessiWay, AudioEye, UserWay и других оверлеев обещают, что достаточно одной строчки JavaScript, чтобы автоматически устранить проблемы с качеством кода, улучшить дизайн интерфейса и привести сайты в _полное_ соответствие с требованиями законов. Компаниям, установившим оверлеи, больше ничего не нужно менять на сайтах. Попробуем разобраться, так ли это на самом деле.

> ### Плюсы и минусы

Author: Пробовали так сделать, получается так, что плюсы только на бумаге, решили объединить вместе всё.

В Штатах около 20% судебных исков касается сайтов с оверлеями, и их количество увеличивается. Например, в 2023 году открыто 449 судебных дел, а 2024 году — 503. Больше статистики по искам в Штатах найдёте на [UsableNet](https://info.usablenet.com/ada-website-compliance-lawsuit-tracker). Уже даже появились юридические фирмы, которые специально ищут сайты с оверлеями, подают в суд на их владельцев, выигрывают дела и получают за это деньги.

**Оверлеи создают угрозу безопасности данных**. Сторонний код на сайте создаёт риск несоблюдения правил хранения и конфиденциальности данных, а ещё не учитывает предпочтения конечных пользователей в объёмах информации, которой они делятся.

Так как поставщики оверлеев могут продавать данные пользователей, появляется риск нарушения Общего регламента по защите персональных данных (GDPR) и Закона Калифорнии о защите персональных данных (CCPA).

**Оверлеи преувеличивают свои возможности**. Поставщики оверлеев преувеличивают количество клиентов и возможности. Этим они создают впечатление, что их продукты популярнее, чем на самом деле.

> <aside>
> 
> 👩‍⚖️ Федеральная торговая комиссия, защищающая права потребителей в США, [оштрафовала accessiBe на 1 миллион долларов](https://www.ftc.gov/news-events/news/press-releases/2025/01/ftc-order-requires-online-marketer-pay-1-million-deceptive-claims-its-ai-product-could-make-websites/) за предоставление ложной гарантии полного соответствия рекомендациям WCAG.
> 
> </aside>

Reviewer 3: ```suggestion
> 👩‍⚖️ Федеральная торговая комиссия, защищающая права потребителей в США, [оштрафовала accessiBe на 1 миллион долларов](https://www.ftc.gov/news-events/news/press-releases/2025/01/ftc-order-requires-online-marketer-pay-1-million-deceptive-claims-its-ai-product-could-make-websites/) за предоставление ложной гарантии полного соответствия рекомендациям WCAG.
```

Почему-то мне кажется, что цитата тут будет выразительнее. Но это вкусовщина, обсуждаемо =)

![CleanShot 2025-01-16 at 11 04 01](https://github.com/user-attachments/assets/7f4e1b80-4601-48ea-b9ee-1976cd99a31a)

![Диалог с умным помощником EqualWeb](images/ai-function.png)

«Умные» оверлеи продолжают ломать пользовательский опыт. Например, они могут сделать интерактивными не те элементы или неправильно описать картинку. Ручное тестирование и исправление багов всё ещё имеют решающее значение для обеспечения максимально возможного уровня доступности сайта. Так что оверлеи не могут заменить экспертов с их пониманием контекста и навыками решения проблем.

**Оверлеи не рекомендуют использовать эксперты**. На громкие заявления производителей оверлеев отреагировало профессиональное сообщество. Более 600 человек из 16 стран подписали заявление против оверлеев «[Overlay Factsheet](https://overlayfactsheet.com/en/)». В списке подписавших письмо не только эксперты по доступности, но и защитники прав людей с инвалидностью, юристы, авторы и редакторы технических спецификаций.

Пользователи доступных интерфейсов тоже критикуют оверлеи. Так, например, Национальная Федерация Незрячих США [призвала запретить использование оверлея AccessiBe](https://nfb.org/about-us/press-room/national-convention-sponsorship-statement-regarding-accessibe/) на американских сайтах.

## Отдельные версии сайтов

> _Отдельная версия сайта_ — это одна или несколько страниц, которые находятся по отдельному URL-адресу. Отдельная версия может отличаться наполнением, например, наличием картинок. Ссылка на такую страницу обычно обозначается иконкой в виде очков и текстом «Аа», «Версия для слабовидящих» и т. д.

Reviewer 3: ```suggestion
_Отдельная версия сайта_ — это одна или несколько страниц, которые находятся по отдельному URL-адресу. Отдельная версия может отличаться наполнением, например, наличием картинок. Ссылка на такую страницу обычно обозначается иконкой в виде очков и текстом «Аа», «Версия для слабовидящих».
```

Особая версия страницы, как правило, — панель с настройками для выбора шрифта, его размера и изменения цветовой темы. Также на отдельной версии сайта отличное от оригинальной страницы расположение содержимого. Это призвано помочь слабовидящим пользователям избежать перегрузки ненужной информацией.

> Пример отдельной версии сайта для пользователей со слепотой и слабовидящих можно увидеть на сайте [Омского портала культуры](https://kultura55.ru).

Reviewer 3: ```suggestion
Пример отдельной версии сайта для пользователей со слепотой и слабовидящих можно увидеть у [Омского портала культуры](https://kultura55.ru).
```

А то «сайта»-«сайте» в одном предложении. Неуютно

На первом скриншоте обычная версия сайта с изображениями в карусели.

![Обычная версия портала культуры](images/ordinary-page.png)

На второй картинке отдельная версия для слабовидящих. На странице нет изображений, увеличен шрифт и установлена чёрно-белая цветовая тема. Дополнительно в отдельной версии сайта можно включить или отключить отрисовку изображений, поменять цветовую тему, величину шрифта и межбуквенное пространство, выбрать тип шрифта (с засечками или без) и изменить межстрочный интервал.

![Отдельная версия портала](images/page-version-for-weak-vision.png)

> ### Достоинства и недостатки

Reviewer 3: Я бы предложила придерживаться схожей с предыдущим пунктом структуры. Переименуем и, может быть, разделим на «Плюсы» и «Минусы»?

Reviewer 2: Там мало плюсов, поэтому сложно вынести в отдельный подраздел.

Первые инструменты для повышения доступности сайтов появились после того, как была завершена работа над первой версией [Руководств по обеспечению доступности веб-контента](/a11y/wcag/) (коротко WCAG). WCAG — основной отраслевой стандарт, определяющий требования к доступности в интернете.

Ранние продукты представляли собой дополнительные элементы управления интерфейсом, зачитывающие содержимое страницы вслух. Разработчики обещали, что такие инструменты приведут к автоматическому соответствию клиентских сайтов рекомендациям WCAG 1.0.

Настройку зачитывания содержимого можно всё ещё встретить на сайтах, в основном, государственных. Обычно переключатель оверлея находится в главном меню и выглядит как иконка с громкоговорителем. При включении оверлея появляется панель управления скоростью, громкостью и порядком чтения, а страница сразу начинает зачитываться.

![Главная страница нидерландского государственного сайта с включённым режимом чтения содержимого](images/reading-overlay.png)

> Главная страница нидерландского государственного сайта с включённым режимом чтения содержимого

Reviewer 2: У картиночек alt и подписи должны быть разными (уникальными в обоих случаях).

> Главная страница нидерландского государственного сайта с включённым режимом чтения содержимого

Reviewer 2: ```suggestion
[ser.nl](https://www.ser.nl/en)
```

Оверлеи позволяют увеличивать размер шрифта, изменять цвет и выравнивание текста, настраивать контрастность цветов или межбуквенное пространство. Примеры дополнительных функций оверлеев — увеличение элементов интерфейса, режим симуляции дальтонизма, выделение отдельных строк текста для людей с дислексией, добавление отсутствующего индикатора фокуса и так далее.

![Некоторые функции и возможности оверлеев](images/overlays.png)

> Некоторые функции и возможности оверлеев

Reviewer 2: ```suggestion
[UserWay](https://userway.org)
```

Поставщики AccessiBe, AccessiWay, AudioEye, UserWay и других оверлеев обещают, что достаточно одной строчки JavaScript, чтобы автоматически устранить проблемы с качеством кода, улучшить дизайн интерфейса и привести сайты в _полное_ соответствие с требованиями законов. Компаниям, установившим оверлеи, больше ничего не нужно менять на сайтах. Так ли это на самом деле?

> ### Плюсы и минусы

Reviewer 2: ```suggestion
### Плюсы и минусы оверлеев
```

Reviewer 2: Можно вот так.

**Оверлеи добавляют ненужные функции**. Многие настройки в оверлеях бесполезны и избыточны. К примеру, режим высокого контраста уже есть в операционных системах. Размер шрифта легко увеличивается или уменьшается во всех браузерах, а режим экранной лупы есть у… экранной лупы. Если пользователи слушают содержимое, то у них есть скринридеры и режимы чтения в браузерах.

Часть функций оверлеев никак не влияет на пользовательский опыт. К примеру, чёрно-белый режим для людей с дальтонизмом. Пользователям, которые не различает все цвета, важнее контрастность текста и элементов.

**Оверлеи непредсказуемы и ненадёжны**. Так как оверлей — это JavaScript-код, на его работу могут влиять интернет-соединение и производительность устройства пользователя. Оверлей может вообще не загрузиться из-за блокировщика рекламы, самого браузера или бага.

Из-за багов в коде, в оверлеях работают не все настройки. К примеру, пользователи, выбрав режим повышенной тёмной контрастности, могут получить чёрный слой поверх всей страницы.

![Неработающая функция режима высокой контрастности в EqualWeb](images/dark-contrast-function.png)

> Неработающая функция режима высокой контрастности в EqualWeb

Reviewer 2: ```suggestion
[EqualWeb](https://www.equalweb.com)
```

Пользователи доступных интерфейсов тоже критикуют оверлеи. Так, например, Национальная Федерация Незрячих США [призвала запретить использование оверлея AccessiBe](https://nfb.org/about-us/press-room/national-convention-sponsorship-statement-regarding-accessibe/) на американских сайтах.

## Отдельные версии сайтов

_Отдельная версия сайта_ — это одна или несколько страниц, которые находятся по отдельному URL-адресу. Отдельная версия может отличаться наполнением, например, наличием картинок. Ссылка на такую страницу обычно обозначается иконкой в виде очков и текстом «Аа», «Версия для слабовидящих».

Особая версия страницы, как правило, — панель с настройками для выбора шрифта, его размера и изменения цветовой темы. Также на отдельной версии сайта отличное от оригинальной страницы расположение содержимого. Это призвано помочь слабовидящим пользователям избежать перегрузки ненужной информацией.

> Пример отдельной версии сайта для пользователей со слепотой и слабовидящих можно увидеть у [Омского портала культуры](https://kultura55.ru).

Reviewer 2: ```suggestion
Пример отдельной версии сайта для пользователей со слепотой и слабовидящих можно увидеть у Омского портала культуры.
```

На первом скриншоте обычная версия сайта с изображениями в карусели.

![Обычная версия портала культуры](images/ordinary-page.png)

> Обычная версия портала культуры

Reviewer 2: ```suggestion
[Главная страница Омского портала культуры](https://kultura55.ru)
```

На второй картинке отдельная версия для слабовидящих. На странице нет изображений, увеличен шрифт и установлена чёрно-белая цветовая тема. Дополнительно в отдельной версии сайта можно включить или отключить отрисовку изображений, поменять цветовую тему, величину шрифта и межбуквенное пространство, выбрать тип шрифта (с засечками или без) и изменить межстрочный интервал.

![Отдельная версия портала](images/page-version-for-weak-vision.png)

> Отдельная версия портала

Reviewer 2: ```suggestion
Всё ещё главная страница портала
```

> ### Плюсы и минусы

Reviewer 2: ```suggestion
### Плюсы и минусы отдельных страниц
```

Reviewer 2: И так.
